src/libpgmoneta/server.c

```diff
@@ -525,6 +525,113 @@ pgmoneta_server_checkpoint(int srv, SSL* ssl, int socket, uint64_t* c_lsn)
    return 1;
 }
 
+int
+pgmoneta_server_file_stat(int srv, SSL* ssl, int socket, char* relative_file_path, struct file_stats* s)
+{
+   char* user = NULL;
+   char* cell_output = NULL;
+   int ret;
+   int q = 0;
+   bool has_privilege = false;
+   char query[MISC_LENGTH];
+   struct message* query_msg = NULL;
+   struct query_response* response = NULL;
+   struct file_stats stat;
```

Comment:
Are you trying to return a memory on on stack? file_stats will be reclaimed after it's out of scope... Instead you should do malloc.

Reply:
Hmm, my mistake. Yes this'll work.